src/coreclr/jit/hwintrinsicxarch.cpp

```diff
@@ -924,7 +924,7 @@ GenTree* Compiler::impNonConstFallback(NamedIntrinsic intrinsic, var_types simdT
             GenTree* op2 = impPopStack().val;
             GenTree* op1 = impSIMDPopStack();
 
-            GenTree* tmpOp = gtNewSimdCreateScalarUnsafeNode(TYP_SIMD16, op2, CORINFO_TYPE_INT, 16);
+            GenTree* tmpOp = gtNewSimdCreateScalarUnsafeNode(simdType, op2, CORINFO_TYPE_INT, genTypeSize(simdType));
```

Comment:
Maybe I misunderstood, but isn't the `TYP_SIMD16` correct for the second arg? For example https://learn.microsoft.com/en-us/dotnet/api/system.runtime.intrinsics.x86.avx512f.shiftrightarithmetic?view=net-8.0#system-runtime-intrinsics-x86-avx512f-shiftrightarithmetic(system-runtime-intrinsics-vector512((system-int32))-system-runtime-intrinsics-vector128((system-int32))) indicates that the second arg is a `TYP_SIMD16` even when the first arg and result is something else.

Reply:
@jakobbotsch you're right these always have 16-byte operand, fixed in `emitGetBaseMemOpSize` where other similar cases are handled